app/internal/page/app.go

```diff
@@ -489,11 +490,28 @@ const (
 	CanGoBack
 )
 
-func makeHandle(mux *http.ServeMux, logger Logger, store sessions.Store, localizer localize.Localizer, lang Lang, rumConfig RumConfig, staticHash string, paths AppPaths, defaultOptions handleOpt) func(string, handleOpt, Handler) {
+func makeHandle(mux *http.ServeMux, logger Logger, store sessions.Store, localizer localize.Localizer, lang Lang, rumConfig RumConfig, staticHash string, paths AppPaths, defaultOptions handleOpt, randomString func(int) string) func(string, handleOpt, Handler) {
 	return func(path string, opt handleOpt, h Handler) {
 		opt = opt | defaultOptions
 
 		mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+			csrfSession, err := store.Get(r, "csrf")
+
+			if r.Method == http.MethodPost {
+				if err != nil {
+					http.Error(w, err.Error(), http.StatusInternalServerError)
+					return
+				}
+
+				if !csrfValid(r, csrfSession) {
+					http.Error(w, "CSRF token not valid", http.StatusForbidden)
+					return
+				}
+			}
+
+			csrfSession.Values = map[interface{}]interface{}{"token": randomString(12)}
```

Comment:
I think we should only set this if not already set, that way you'll still be able to press the back button and submit a form again.

Reply:
Yeah, I had thought about setting this once per period of time/session but I wasn't sure if this made things less secure. Happy to keep it around for a day and this will hopefully fix the cypress visitLpaPath weirdness I was seeing.